packages/ember-routing/lib/system/router.js

```diff
@@ -987,6 +1032,8 @@ function calculatePostTransitionState(emberRouter, leafRouteName, contexts) {
 
 function updatePaths(router) {
   let infos = router.router.currentHandlerInfos;
+  if (infos.length === 0) { return; }
```

Comment:
I'm running into a similar problem but in my case `infos` isn't even an array.  I did something like:

```
import { isArray } from 'ember-runtime/utils';
...
if (!isArray(infos) || infos.length === 0) { return; }
```